### ISSUE TITLE
[Error]ArgumentException: Off-axis dimensions must match when using a pre-trained Inception-V3 model.

### ISSUE THREAD

**Issue by JetttZHENG** [Unity-Technologies/ml-agents#2030]
Hello, guys, I would like to try pre-trained Keras Model (Inception-V3) with Barracuda, Error occurring, could someone help? Many thanks. Details following:

1. I have a pre-trained Keras model (Inception-V3 model from Keras API).

2. converted the h5 model to pb model using a third-party tool (no errors occurring)

3. converted the pb model to nn model using tensorflow_to_barracuda. (no errors occurring), changed its name from model.nn to model.bytes.

4. followed Barracuda.md, I created a script to get the output of the model.bytes model. Then following error occurred.
![image](https://user-images.githubusercontent.com/43242747/57569594-d0f97000-7429-11e9-8a10-1fc7109b8786.png)

All models and tool can be found in this link:
https://drive.google.com/drive/folders/1l4hMasPIKYhV9UbMogHqVQE6fbS4MhnG?usp=sharing

**Comment by JetttZHENG** [comment]
`using System.Collections;
using System.Collections.Generic;
using UnityEngine;
using MLAgents;
using Barracuda;
using System.IO;
using UnityEngine.UI;

public class TestTensorflow2 : MonoBehaviour
{
    private string loadpath = "Assets/Materials/imagenet_test.png";
    // Start is called before the first frame update
    void Start()
    {
        Tensor output_nn = InferenceTest(loadpath);

    }

    private Tensor InferenceTest(string loadpath)
    {
        Texture2D input_texture2D = LoadByIO(loadpath);
        // Color[] picture_color = input_texture2D.GetPixels();

        var inputs = new Dictionary<string, Tensor>();

        // Load Model into Barracuda
        var model = ModelLoader.LoadFromStreamingAssets("Inception-V3_1" + ".bytes");

        // Create inference engine (Worker)
        var worker = BarracudaWorkerFactory.CreateWorker(BarracudaWorkerFactory.Type.ComputeFast, model);

        //Create inputs
        Tensor tensor = new Tensor(input_texture2D, 3);
        tensor.Print();
        Debug.Log(tensor[0, 12, 12, 0]);
        inputs.Add("input_3", tensor);

        //Execute the model
        worker.Execute(inputs);

        //Fetch outputs
        var output_nn = worker.Fetch();

        //output_nn.Print();
        //Debug.Log(output_nn);

        worker.Dispose();
        return output_nn;
    }



    private Texture2D LoadByIO(string loadpath)
    {
        //FileStream creat
        FileStream fileStream = new FileStream(loadpath, FileMode.Open, FileAccess.Read);
        fileStream.Seek(0, SeekOrigin.Begin);
        //
        byte[] bytes = new byte[fileStream.Length];
        //FileStream read
        fileStream.Read(bytes, 0, (int)fileStream.Length);
        //FileStream close
        fileStream.Close();
        fileStream.Dispose();
        fileStream = null;

        // Ceate Texture
        int width = 299;
        int height = 299;
        Texture2D texture2D = new Texture2D(width, height);
        var is_load = texture2D.LoadImage(bytes);
        if (!is_load)
        {
            Debug.Log("Texture Fail");
        }
        return texture2D;
    }
    // Update is called once per frame
    void Update()
    {

    }
}
`

**Comment by KozykinRoman** [comment]
Hi, It seems you didn't use any output node in Fetch function.
So plz check for it with tensorboard.

And on the other hand, I really want to know how to get trained model using Keras?
Of course it will be better to teach me on teamviewer or skype, but it is also fine to give me the reference url link for it. I couldn't find any proper video tutorial for that.
Thanks. Kozykin.

**Comment by ervteng** [comment]
@mantasp

**Comment by JetttZHENG** [comment]
Sure, I will try it and let you know whether it works. Thanks!


Zheng Zhenjie

**Comment by JetttZHENG** [comment]
Hi @mantasp,  I have retried the Inception-V3 model with the barracuda-0.2.0 . 

No error message occurred this time, but the output tensor are all NAN that are not consistent with  the Keras model results.

There might be some something lost when converting the model or bugs with barracuda-0.2.0.

Any suggestion? Thanks!

**Comment by mantasp** [comment]
@JetttZHENG how are you converting keras to tf? In the past we had pretty good experience using this converter: https://github.com/amir-abdi/keras_to_tensorflow/blob/master/keras_to_tensorflow.py

**Comment by JetttZHENG** [comment]
@mantasp The same converter.

And the Keras model came directly using the Keras API:
model = keras.applications.inception_v3.InceptionV3(include_top=True, weights='imagenet', input_tensor=None, input_shape=(299,299,3), pooling=None, classes=1000);

**Comment by mantasp** [comment]
@JetttZHENG which Keras applications version are you using?

**Comment by xiaomaogy** [comment]
Thanks for reaching out to us. Hopefully you were able to resolve your issue.  We are closing this due to inactivity, but if you need additional assistance, feel free to reopen the issue.
